compiler/parser.nim

```diff
@@ -2031,12 +2033,15 @@ proc parseObjectPart(p: var Parser): PNode =
     result = p.emptyNode
 
 proc parseObject(p: var Parser): PNode =
-  #| objectDecl = 'object' pragma? ('of' typeDesc)? COMMENT? objectPart
+  #| objectDecl = 'object' ('of' typeDesc)? COMMENT? objectPart
   result = newNodeP(nkObjectTy, p)
   getTok(p)
   if p.tok.tokType == tkCurlyDotLe and p.validInd:
     # Deprecated since v0.20.0
-    parMessage(p, warnDeprecated, "type pragmas follow the type name; this form of writing pragmas is deprecated")
+    parMessage(p, warnDeprecated,
+      "pragmas after the `object` keyword is a " &
+      "legacy syntax kept for compatibility; " &
+      "put pragmas after the type name instead")
```

Comment:
So does a pragma after `object` still work or not? This warning makes it sound like it still works ("kept for compatibility"). If it doesn't work anymore, why is this just a warning instead of a hard error?

Reply:
It was originally a hard error but I ended up just changing the warnings because I thought it might be annoying to update packages. I forgot to update the changelog accordingly, don't know the best phrasing anyway. It might be better to just remove it, but to do so either a PR has to be made to ast_pattern_matching on [this line](https://github.com/krux02/ast-pattern-matching/blob/1a866d2e536541c032a50323f36bb482df9e3b39/src/ast_pattern_matching.nim#L575) for CI to pass since nimlsp depends on it, or temporarily disable nimlsp which isn't great.

Reply:
So now the changelog is wrong, since it isn't actually removed.